sources/packages/api/src/route-controllers/institution/institution.controller.ts

```diff
@@ -259,4 +264,50 @@ export class InstitutionController extends BaseController {
       body.isActive,
     );
   }
+
+  @Get("/my-details")
+  async getMyDetails(
+    @UserToken() token: IInstitutionUserToken,
+  ): Promise<InstitutionUserAndAuthDetails> {
+    // Get logged in user and location details with auth
+    const institutionUser =
+      await this.institutionService.getInstitutionUserByUserName(
+        token.userName,
+      );
+    const user = {
+      user: {
+        firstName: institutionUser.user?.firstName,
+        lastName: institutionUser.user?.lastName,
+        userName: token.userName ?? institutionUser.user?.userName,
+        isActive: institutionUser.user?.isActive,
+        isAdmin: token.authorizations.isAdmin,
+        id: institutionUser.user?.id,
+        email: token.email ?? institutionUser.user?.email,
+      },
+    };
+    const LocationAuth = {
+      authorizations: token.authorizations,
+    };
+
+    return {
+      ...user,
+      ...LocationAuth,
```

Comment:
Please consider not exposing the InstitutionUserAuthorizations directly in the API.
In the future, if someone adds some sensitive data to InstitutionUserAuthorizations it could be automatically exposed by mistake. That's why we try to enforce the DTOs' creation to ensure that future maintenance and changes do not expose more than what API is supposed to do.